scipy/sparse/linalg/isolve/lsmr.py

```diff
@@ -221,6 +221,11 @@ def lsmr(A, b, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8,
     if maxiter is None:
         maxiter = minDim
 
+    if x0 is None:
+        dtype = result_type(A, b, float)
+    else:
+        dtype = result_type(A, b, x0, float)
```

Comment:
Actually one could always do this, because `None` behaves the same as `float`. But maybe that's confusing?

Reply:
Better to be explicit here, I think.